test/e2e/kube-ovn/underlay/underlay.go

```diff
@@ -327,4 +340,276 @@ var _ = framework.Describe("[group:underlay]", func() {
+			} else {
+				framework.ExpectEqual(int(subnet.Status.V4UsingIPs), 1)
+			}
+
+			agStr := strings.Replace(fmt.Sprintf("%s.u2o_exclude_ip.ip4", subnet.Name), "-", ".", -1)
+			ginkgo.By("checking underlay subnet's policy1 route")
+			hitPolicyStr := fmt.Sprintf("[31000 ip4.dst == $%s && ip4.src == %s allow]", agStr, cidr)
+			checkPolicy(hitPolicyStr, isEnableU2O)
+
+			ginkgo.By("checking underlay subnet's policy2 route")
+			hitPolicyStr = fmt.Sprintf("[31000 ip4.dst == %s && ip4.dst != $%s allow]", cidr, agStr)
+			checkPolicy(hitPolicyStr, isEnableU2O)
+
+			ginkgo.By("checking underlay subnet's policy3 route")
+			hitPolicyStr = fmt.Sprintf("[29000 ip4.src == %s reroute %s]", cidr, v4gw)
+			checkPolicy(hitPolicyStr, isEnableU2O)
+		} else {
+			if isEnableU2O {
+				framework.ExpectEqual(int(subnet.Status.V6UsingIPs), 2)
+			} else {
+				framework.ExpectEqual(int(subnet.Status.V6UsingIPs), 1)
+			}
+
+			agStr := strings.Replace(fmt.Sprintf("%s.u2o_exclude_ip.ip6", subnet.Name), "-", ".", -1)
+			ginkgo.By("checking underlay subnet's policy1 route")
+			hitPolicyStr := fmt.Sprintf("[31000 ip6.dst == $%s && ip6.src == %s allow]", agStr, cidr)
+			checkPolicy(hitPolicyStr, isEnableU2O)
+
+			ginkgo.By("checking underlay subnet's policy2 route")
+			hitPolicyStr = fmt.Sprintf("[31000 ip6.dst == %s && ip6.dst != $%s allow]", cidr, agStr)
+			checkPolicy(hitPolicyStr, isEnableU2O)
+
+			ginkgo.By("checking underlay subnet's policy3 route")
+			hitPolicyStr = fmt.Sprintf("[29000 ip6.src == %s reroute %s]", cidr, v6gw)
+			checkPolicy(hitPolicyStr, isEnableU2O)
+		}
+	}
+
+	ginkgo.By("checking underlay pod's ip route's nexthop equal the u2o interconnection ip")
+	routes, err := getPodDefaultRoute(underlayPod)
+	framework.ExpectNoError(err)
+	framework.ExpectNotEmpty(routes)
+
+	interconnIps := strings.Split(subnet.Status.U2OInterconnectionIP, ",")
+	var v4InterconnIp, v6InterconnIp string
+	for _, connIp := range interconnIps {
+		if util.CheckProtocol(connIp) == "IPv4" {
+			v4InterconnIp = connIp
+		} else {
+			v6InterconnIp = connIp
+		}
+	}
+
+	for _, route := range routes {
+		if route.Dst == "default" {
+			if util.CheckProtocol(route.Gateway) == "IPv4" {
+				if isEnableU2O {
+					framework.ExpectEqual(route.Gateway, v4InterconnIp)
+				} else {
+					framework.ExpectEqual(route.Gateway, v4gw)
+				}
+			} else {
+				if isEnableU2O {
+					framework.ExpectEqual(route.Gateway, v6InterconnIp)
+				} else {
+					framework.ExpectEqual(route.Gateway, v6gw)
+				}
+			}
+		}
+	}
+
+	ginkgo.By("checking underlay pod access to overlay pod")
+	checkPing(underlayPod.Name, underlayPod.Namespace, overlayPod.Status.PodIP, isEnableU2O)
+}
+
+func getPodDefaultRoute(pod *corev1.Pod) ([]iproute.Route, error) {
+	var routes, routes6 []iproute.Route
+	stdout, _ := exec.Command("bash", "-c", fmt.Sprintf("kubectl exec %s -n %s -- ip -d -j route show dev eth0", pod.Name, pod.Namespace)).CombinedOutput()
```

Comment:
Use functions defined in `test/e2e/framework/iproute/iproute.go`.

Reply:
it seems RouteShow in test/e2e/framework/iproute/iproute.go only show the route in node but not in pod

Reply:
https://github.com/kubeovn/kube-ovn/blob/master/test/e2e/kube-ovn/underlay/underlay.go#L349